app/dashboard/views.py

```diff
@@ -4044,3 +4047,103 @@ def choose_persona(request):
             'persona': persona,
         },
         status=200)
+
+
+@csrf_exempt
+@require_POST
+def join_tribe(request, handle):
+    print(handle, request.user.profile)
+    if request.user.is_authenticated:
+        profile = request.user.profile if hasattr(request.user, 'profile') else None
+        print(profile)
+        try:
```

Comment:
Lets not use `try` / `except` here - if something goes wrong (like a user doesn't have a profile, or the handle doesn't exist) then its going to jump to the exception handling code. Just use an if statement here with exists to see if the user is a tribe member like `if profile and handle and TribeMember.objects.get(profile=profile, ...).exists(): TribeMember.objects.get(...).delete()`

Reply:
I will get back this to try because is failing in this way
`File "/code/app/dashboard/views.py", line 4057, in join_tribe
web_1        |     if profile and handle and TribeMember.objects.get(profile=profile, org__handle__iexact=handle).exist():
web_1        | dashboard.models.TribeMember.DoesNotExist: TribeMember matching query does not exist.`